### ISSUE TITLE
Be concrete about && and || at the beginning or end of a line

### ISSUE THREAD

**Issue by vRallev** [android/kotlin-guides#62]
I think Java prefers
```java
if (condition1
    && condition 2
    || condition3) {

    // do something
}
```
This style would make sense to me in Kotlin as well, but I couldn't find anything regarding to that in the IJ Kotlin style guide or Android style guide. Note that there has been some discussion going on in the ktlint project what prefers
```kotlin
if (condition1 &&
    condition2 ||
    condition3) {

    // do something
}
```
https://github.com/shyiko/ktlint/issues/168

**Comment by JakeWharton** [comment]
Tragically (in my opinion), Kotlin forces you into the latter as otherwise things like
```kotlin
foo
+ bar
```
could be interpreted as
```kotlin
foo
unaryPlus(bar)
```
depending on what is in scope.

I thought we covered this somewhere but perhaps it was left ambiguous. Would be good to include.

**Comment by JakeWharton** [comment]
That example doesn't actually apply to the binary boolean operators, but I see no reason to create archaic rules around _which_ binary operators and types you're operating on so we'll normalize to end-of-line rather than start-of-line.

**Comment by vRallev** [comment]
I agree that it's really sad. At least I have something else besides ktlint that I could reference. Thanks
